modules/33acrossBidAdapter.js

```diff
@@ -173,53 +245,187 @@ function _createServerRequest({bidRequest, gdprConsent = {}, uspConsent, pageUrl
     'options': options
   }
 }
 
-// Sync object will always be of type iframe for TTX
-function _createSync({ siteId = 'zzz000000000003zzz', gdprConsent = {}, uspConsent }) {
-  const ttxSettings = config.getConfig('ttxSettings');
-  const syncUrl = (ttxSettings && ttxSettings.syncUrl) || SYNC_ENDPOINT;
+// BUILD REQUESTS: SIZE INFERENCE
+function _transformSizes(sizes) {
+  if (utils.isArray(sizes) && sizes.length === 2 && !utils.isArray(sizes[0])) {
+    return [ _getSize(sizes) ];
+  }
 
-  const { consentString, gdprApplies } = gdprConsent;
+  return sizes.map(_getSize);
+}
 
-  const sync = {
-    type: 'iframe',
-    url: `${syncUrl}&id=${siteId}&gdpr_consent=${encodeURIComponent(consentString)}&us_privacy=${encodeURIComponent(uspConsent)}`
-  };
+function _getSize(size) {
+  return {
+    w: parseInt(size[0], 10),
+    h: parseInt(size[1], 10)
+  }
+}
 
-  if (typeof gdprApplies === 'boolean') {
-    sync.url += `&gdpr=${Number(gdprApplies)}`;
+// BUILD REQUESTS: PRODUCT INFERENCE
+function _getProduct(bidRequest) {
+  const { params, mediaTypes } = bidRequest;
+
+  const { banner, video } = mediaTypes;
+
+  if ((video && !banner) && video.context === 'instream') {
+    return PRODUCT.INSTREAM;
   }
 
-  return sync;
+  return (params.productId === PRODUCT.INVIEW) ? (params.productId) : PRODUCT.SIAB;
 }
 
-function _getBidFloors(getFloor, size) {
-  const bidFloors = getFloor({
+// BUILD REQUESTS: BANNER
+function _buildBannerORTB(bidRequest) {
+  const bannerAdUnit = utils.deepAccess(bidRequest, 'mediaTypes.banner', {});
+  const element = _getAdSlotHTMLElement(bidRequest.adUnitCode);
+
+  const sizes = _transformSizes(bannerAdUnit.sizes);
+
+  let format;
+
+  // We support size based bidfloors so obtain one if there's a rule associated
+  if (typeof bidRequest.getFloor === 'function') {
+    format = sizes.map((size) => {
+      const bidfloors = _getBidFloors(bidRequest, size, BANNER);
+
+      let formatExt;
+      if (bidfloors) {
+        formatExt = {
+          ext: {
+            ttx: {
+              bidfloors: [ bidfloors ]
+            }
+          }
+        }
+      }
+
+      return Object.assign({}, size, formatExt);
+    });
+  } else {
+    format = sizes;
+  }
+
+  const minSize = _getMinSize(sizes);
+
+  const viewabilityAmount = _isViewabilityMeasurable(element)
+    ? _getViewability(element, utils.getWindowTop(), minSize)
+    : NON_MEASURABLE;
+
+  const ext = contributeViewability(viewabilityAmount);
+
+  return {
+    format,
+    ext
+  }
+}
+
+// BUILD REQUESTS: VIDEO
+// eslint-disable-next-line no-unused-vars
+function _buildVideoORTB(bidRequest) {
+  const videoAdUnit = utils.deepAccess(bidRequest, 'mediaTypes.video', {});
+  const videoBidderParams = utils.deepAccess(bidRequest, 'params.video', {});
+
+  const videoParams = {
+    ...videoAdUnit,
+    ...videoBidderParams // Bidder Specific overrides
+  };
+
+  const video = {}
+
+  const {w, h} = _getSize(videoParams.playerSize[0]);
```

Comment:
@msm0504 assuming that playerSize although set as an `[int, int]` in Ad Unit definition (per docs) will be sent as `[[int, int]]` which is how I am seeing in `bidRequest`

Reply:
That's correct